spec/helper_spec.rb

```diff
@@ -41,9 +41,26 @@
     end
 
     context "payload length is greater than allowed" do
+      it "trims exception messages" do
+        payload = {
+          :events => [{
+            :exceptions => [{
+              :message => 50000.times.map {|i| "should truncate" }.join(""),
+              :preserved => "Foo"
+            }]
+          }]
+        }
+        expect(::JSON.dump(payload).length).to be > Bugsnag::Helpers::MAX_PAYLOAD_LENGTH
+        trimmed = Bugsnag::Helpers.trim_if_needed(payload)
+        expect(::JSON.dump(trimmed).length).to be <= Bugsnag::Helpers::MAX_PAYLOAD_LENGTH
+        expect(trimmed[:events][0][:exceptions][0][:message].length).to be <= Bugsnag::Helpers::MAX_STRING_LENGTH
```

Comment:
This test doesn't verify what the value is, in case its been converted to an empty string or shortened more than needed or something.

Reply:
Improved the assertions in https://github.com/bugsnag/bugsnag-ruby/pull/646/commits/0a2101724193d17af7b120d75d939f7295d1c90a